src/device-state.coffee

```diff
@@ -675,24 +676,30 @@ module.exports = class DeviceState extends EventEmitter
 		@applyUnblocker?()
 		return
 
-	triggerApplyTarget: ({ force = false, delay = 0, initial = false } = {}) =>
-		if @applyInProgress
-			if !@scheduledApply?
+	triggerApplyTarget: ({ force = false, delay = 0, initial = false, isFromApi = false } = {}) =>
+		if @applyInProgress?
+			if !@scheduledApply? || (isFromApi && @cancelDelay)
 				@scheduledApply = { force, delay }
+				if isFromApi
+					# Cancel promise delay if call came from api to prevent waiting due to backoff
+					@cancelDelay()
 			else
 				# If a delay has been set it's because we need to hold off before applying again,
 				# so we need to respect the maximum delay that has been passed
 				@scheduledApply.delay = Math.max(delay, @scheduledApply.delay)
 				@scheduledApply.force or= force
 			return
-		@applyInProgress = true
-		Promise.delay(delay)
+		@applyInProgress = new Promise (resolve, reject) =>
```

Comment:
Now that there's no cancellation, I think there's no need to assign the Promise to `@applyInProgress` anymore, so this could just be:

```
@applyInProgress = true
new Promise (resolve, reject) =>
 	setTimeout(resolve, delay)
	@cancelDelay = reject
.then =>
	@cancelDelay = null
	...
.finally =>
	@applyInProgress = false
	...
```

(But this doesn't hurt anyways so just a suggestion)

Reply:
Oh yeah you are right. We don't need the promise reference anymore. I've reverted to using booleans for applyInProgress again. Thanks for the input.